extensions/spring-cache/deployment/src/main/java/io/quarkus/spring/cache/SpringCacheUtil.java

```diff
@@ -0,0 +1,48 @@
+package io.quarkus.spring.cache;
+
+import java.util.Optional;
+
+import org.jboss.jandex.AnnotationInstance;
+import org.jboss.jandex.AnnotationTarget;
+import org.jboss.jandex.AnnotationValue;
+
+final class SpringCacheUtil {
+
+    private SpringCacheUtil() {
+    }
+
+    /**
+     * Meant to be called for instances of {@code @Cacheable}, {@code @CacheEvict} or {@code @CachePut}
+     * Returns the single name of the cache to use
+     */
+    static Optional<String> getSpringCacheName(AnnotationInstance annotationInstance) {
+        String cacheName = null;
+        AnnotationValue cacheNameValue = annotationInstance.value("cacheNames");
+        if (cacheNameValue != null) {
+            cacheName = singleName(cacheNameValue, annotationInstance.target());
+        }
+        if (cacheName == null || cacheName.isEmpty()) {
+            AnnotationValue value = annotationInstance.value();
+            if (value != null) {
+                cacheName = singleName(value, annotationInstance.target());
+            }
+        }
+        return cacheName == null ? Optional.empty() : Optional.of(cacheName);
```

Comment:
You had an empty string check before, it is gone on purpose?

I expected `return cacheName == null || cacheName.isEmpty() ? Optional.empty() : Optional.of(cacheName);` here.

Reply:
Yeah I guess you are right. I should update it to be on the safe side

Reply:
Fixed